File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.html

#Oczliczanie potegi podstawy podniesionej do wykładnika.

def potega_it(a, n):
    wynik = 1
    for i in range(n):
        
        
        wynik = wynik * a
        #print(wynik)
    return wynik
    
    



def main(args):
    #a = int(input("pdodaj podstawe:"))
    #n = int(input("pdodaj podstawe:"))
    #print("Potega {} do {} wynosi {}". format(a, n, potega_it(a, n)))
    assert(potega_it(1, 1) == 1)
    assert(potega_it(2, 10) == 1024)
    assert(potega_it(1, 0) == 1)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cxx
/*
 * potega.cxx
 * 
 * Copyright 2018  <>
 
 */


#include <iostream>
using namespace std;

int  potega_it(int a, int n){
    int wynik = 1;
    for(int i = 0; i < n; i++)
     {
        
        wynik = wynik * a;
        
        
        }
        return wynik;
    
    
}
    
int potega_re(int a, int n)
    {
        if (n== 0) return 1;
        return potega_re(a, n-1) * a;
            
    
    
    
    }    



int main(int argc, char **argv)
{
    int a, n;
    cout << "Podaj podstawe i wykładnik";
    cin >> a >> n;
    cout << "potega:" << potega_it(a, n) << endl;
    cout << "potega:" << potega_re(a, n) << endl;
	
	return 0;
}

